ckanext/versioning/logic/action.py

```diff
@@ -113,6 +128,17 @@ def dataset_version_create(context, data_dict):
         raise toolkit.ValidationError(
             'Version names must be unique per dataset'
         )
+    # TODO: Names like 'Version 1.2' are not allowed as Github tags
```

Comment:
If you want to pass validation to the backend, I suppose you can do a `try` / `except ValueError` around the `tag_create` call. That would work but may not provide the best UX as CKAN level validation.